hail/hail/src/is/hail/expr/ir/lowering/LoweringPass.scala

```diff
@@ -48,6 +48,11 @@ case class OptimizePass(_context: String) extends LoweringPass {
   override val context = s"Optimize: ${_context}"
   override val before: Invariant = AnyIR
   override val after: Invariant = AnyIR
+
+  override def apply(ctx: ExecuteContext, ir: BaseIR): BaseIR =
+    if (ctx.flags.isDefined(Optimize.Flags.Optimize)) super.apply(ctx, ir)
+    else ir
```

Comment:
I'm a little sad that I can't see a way to do this without overriding apply here. It would be nice if we could keep the apply method final.